### PR TITLE
Support migrate assessments machines

### PR DESCRIPTION
**Pull request by sathish-progress**
### Description
Support Azure Migrate Assessment Machine(s)

### Check List

- [X] New functionality includes tests
- [X] All tests pass
- [X] `rake lint` passes
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: docs/resources/azure_migrate_assessment_machine.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+## Azure REST API version, endpoint and http client parameters
+
+This resource interacts with api versions supported by the resource provider.
+The `api_version` can be defined as a resource parameter.
+If not provided, the latest version will be used.
+For more information, refer to [`azure_generic_resource`](azure_generic_resource.md).
+
+Unless defined, `azure_cloud` global endpoint, and default values for the http client will be used.
+For more information, refer to the resource pack [README](../../README.md).
+
```
Comment (sa-progress) [inline docs/resources/azure_migrate_assessment_machine.md:19]: Please add the link for this resource api.

File: docs/resources/azure_migrate_assessment_machine.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+## Availability
+
+### Installation
+
+This resource is available in the [InSpec Azure resource pack](https://github.com/inspec/inspec-azure).
+For an example `inspec.yml` file and how to set up your Azure credentials, refer to resource pack [README](../../README.md#Service-Principal).
+
```
Comment (sa-progress) [inline docs/resources/azure_migrate_assessment_machine.md:26]: Please add the link for this resource api.

File: docs/resources/azure_migrate_assessment_machine.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+```
+## Parameters
+
+| Name           | Description                                                                      |
+|----------------|----------------------------------------------------------------------------------|
+| name           | Name of the Azure Migrate Assessment Machine to test.                                   |
+| resource_group | Azure resource group that the targeted resource resides in. `MyResourceGroup`    |
+| project_name   | Azure Migrate Assessment Project.                                                |
+| group_name     | Unique name of a group within a project.                                         |
+| name           | Unique name of an assessment within a project.                                   |
```
Comment (sa-progress) [inline docs/resources/azure_migrate_assessment_machine.md:52]: Please verify the params.

**Review by sa-progress — CHANGES_REQUESTED** [review]
I Don't see terraform changes .Please check once.

File: docs/resources/azure_migrate_assessment_machine.md (hunk 2 above)
Comment (sathish-progress) [reply docs/resources/azure_migrate_assessment_machine.md:26]: done..

**Comment by sathish-progress** [comment]
> I Don't see terraform changes .Please check once.

No terraform available for this resource

**Review by sa-progress — APPROVED** [review]
LGTM

**Review by IanMadd — APPROVED** [review]

**Pull request merged**